### PR TITLE
v3.34.83 — STRK-99 hotfix: fix BE spot-ticker / sidebar leak

### PR DESCRIPTION
**Pull request by lbruton**
> **Draft — QA preview.** Merge to `dev` after CI + review.

## Summary

Hotfix for [STRK-99](https://plane.lbruton.cc/lbruton/projects/026dbe54-fe52-4a9f-9f1b-7edcb9bbdceb/issues/138ff49f-cfde-4efc-8755-99616f76aae8). v3.34.82's denylist correctly blocked SDB and BE from getting their bulk-tier JSON-LD price recorded — but then routed both vendors through the markdown extractor, where Byparr's React shell returned **only the spot ticker** in `<header>` for BE. `firstInRangePriceProse()` matched the ticker, so the dashboard showed silver $75.92 / gold $4,520 for every BE coin.

## Two-pronged fix

**1.** `htmlToPlainText()` drops `<nav>`, `<header>`, `<footer>` blocks the same way it already drops `<script>`/`<style>`. New `CHROME_OR_RAWTEXT_TAGS` Set makes the skip-list explicit. HTML5 forbids `<main>`-style product content inside these tags, so the strip is safe across every vendor using the cf-clearance HTML path.

**2.** New `tierAnchoredPrice()` helper matches qty-range prose patterns (`1-24`, `1 - 49`, `50+`, `1+`) and returns the first wire price after the first match. `UNTRUSTED_OFFER_PRICE_VENDORS` (sdbullion, bullionexchanges) now resolve via pipe-table → tier-anchored only; `firstInRangePriceProse` is **no longer reachable** for these vendors, so spot tickers and "Related Products" sidebar prices can never poison the dashboard again. Returns `null` when neither matcher hits — caller treats that as OOS rather than recording a misleading value.

## Tests

| Suite | New | Total |
|---|---|---|
| `price-extract-jsonld.test.mjs` (STRK-99 v1) | — | 25 ✅ |
| `price-extract-html-chrome-strip.test.mjs` (this PR) | 12 | 12 ✅ |
| `price-extract-tier-anchored.test.mjs` (this PR) | 18 | 18 ✅ |
| **Total** | **30** | **55 ✅** |

## In-container live verification (DRY_RUN against home-poller Byparr)

| Vendor + Coin | Path | Price | Status |
|---|---|---|---|
| `sdbullion / maple-silver` | `playwright-direct:tierAnchored` | **$79.23** | ✅ correct 1-unit Check/Wire |
| `sdbullion / maple-gold` | `playwright-direct:tierAnchored` | **$4,580.63** | ✅ correct 1-unit |
| `bullionexchanges / maple-silver` | `cf-clearance:tierAnchored` | **$77.82** | ✅ correct 1-unit Check/Wire |
| `bullionexchanges / maple-gold` | exhausted | NULL | ✅ safe-OOS (Byparr couldn't hydrate the price grid in its render window — better than recording spot or sidebar) |

Overall scrape: 17/18 prices captured, 1 NULL (BE gold maple — accepted as OOS per design).

## Bug caught DURING in-container test

The unit-test fixture used `1-24` (no spaces around dash). SDB renders qty as `1 - 49` (whitespace). The v1 regex silently skipped the first match and landed on the bulk tier — recording $78.23 instead of $79.23 for SDB silver maple. v2 regex accepts either form AND caps the LOW side at 3 digits so year ranges like `2024 - 2025` can't false-match. New test `1a` locks in the SDB shape; `12a` / `12b` lock in the year-range guard. **Worth a permanent reflex: regex extractors must be validated against captured real text from the target, not just synthetic fixtures.**

## Follow-up filed separately

Per-vendor / per-item extractor granularity (user's request): each vendor in its own module with per-URL/per-item extractor functions, so tuning one product doesn't risk regressing another. Tracking in next STRK issue (filed after this PR merges).

## Summary by CodeRabbit

* **Bug Fixes**
  * Fixed price extraction accuracy for certain vendors by filtering misleading spot tickers and sidebar promotional content.
  * Improved handling of quantity-tier pricing patterns for more reliable price detection.

* **Chores**
  * Version bumped to 3.34.83.

[Review Change Stack](https://app.coderabbit.ai/change-stack/lbruton/StakTrakr/pull/1148?utm_source=github_walkthrough&utm_medium=github&utm_campaign=change_stack)

### CODE REVIEW

File: devops/pollers/shared/price-extract.js [hunk 1]
```diff
@@ -966,7 +1032,16 @@ function htmlToPlainText(html) {
...
+    // STRK-99 hotfix: drop entire <script>, <style>, AND SPA chrome blocks
+    // (<nav>, <header>, <footer>) before flattening to plain text. The chrome
+    // tags carry site-wide spot tickers and mega-menu links that otherwise
+    // poison firstInRangePriceProse() on vendors whose product price is JS-
+    // hydrated AFTER Byparr's render window — bullionexchanges was returning
+    // spot silver/gold from the header ticker for every coin because Byparr
+    // captured the React shell before the product price grid rendered.
+    // HTML5 forbids <main>-style product content inside these tags, so this
+    // is safe across all vendors using the cf-clearance HTML path.
+    if (!isClosingTag && CHROME_OR_RAWTEXT_TAGS.has(tagName)) {
```
[the remark this answers is not in the document]
Comment (lbruton) [reply devops/pollers/shared/price-extract.js]: Good catch — applied in be26e3b2. Added a `isCompleteTagName` guard that requires the char after tagName to be HTML whitespace, `>`, or `/` before treating it as a real chrome tag. Without it, `<nav-bar>` would have been mis-identified as `<nav>` and `findRawTextCloseTag` failing to find `</nav>` would have silently dropped the rest of the document. 4 new regression tests in the chrome-strip suite (`<nav-bar>`, `<header-banner>`, `<footer1>`, and the mixed real-`nav` + custom-`nav-bar` case) all green; full suite 16/16.

File: devops/pollers/shared/price-extract-html-chrome-strip.test.mjs [hunk 2]
```diff
@@ -0,0 +1,256 @@
...
+  let i = from;
+  while (i < html.length) {
+    const idx = lowerHtml.indexOf(closeTag, i);
+    if (idx === -1) return -1;
+    const after = idx + closeTag.length;
+    if (after >= html.length || /[\s>/]/.test(html[after])) return findHtmlTagEnd(html, after);
+    i = after;
+  }
+  return -1;
+}
```
[the remark this answers is not in the document]
Comment (lbruton) [reply devops/pollers/shared/price-extract-html-chrome-strip.test.mjs:52]: Looked at this closely — the two `findRawTextCloseTag` impls are functionally equivalent for well-formed HTML but structured differently on purpose. The production version walks chars defensively (handles `< / nav >` with whitespace around the close marker, validates name slice equality); the test version uses an `indexOf` shortcut with a trailing-char regex. Production is intentionally more conservative — backporting the simpler test version would actually remove the whitespace-tolerant close-tag matching. Leaving production as-is; opting to keep the test helper slim for readability. If you want them perfectly symmetric, the right move is to import the production helper from price-extract.js rather than duplicate either implementation, but that would require exporting it (the file currently exports nothing). Happy to do that as a follow-up if you think the duplication is worth eliminating.

File: devops/pollers/shared/price-extract-html-chrome-strip.test.mjs [hunk 3]
```diff
@@ -0,0 +1,256 @@
...
+    if (isClosingTag) nameStart++;
+    while (nameStart < html.length && isHtmlWhitespace(html[nameStart])) nameStart++;
+    let nameEnd = nameStart;
+    while (nameEnd < html.length) {
+      const code = lowerHtml.charCodeAt(nameEnd);
+      if (!((code >= 97 && code <= 122) || (code >= 48 && code <= 57))) break;
+      nameEnd++;
+    }
+    const tagName = lowerHtml.slice(nameStart, nameEnd);
+    if (!isClosingTag && CHROME_OR_RAWTEXT_TAGS.has(tagName)) {
```
[the remark this answers is not in the document]
Comment (lbruton) [reply devops/pollers/shared/price-extract-html-chrome-strip.test.mjs]: Same fix mirrored in the inlined test copy in be26e3b2 — added the `isCompleteTagName` guard so the test helper stays in lockstep with the production behavior.
